README.md

```diff
@@ -287,53 +287,161 @@ Although this example is pretty simple and not even includes Lens composition, i
 
 Just as with React, but assuming Angular instead :)
 
+The code for below example - can be found here <https://github.com/georgebatalinski/telescope_todo>
+
 As a bootstrap for the application we need something like:
 
+```typescript 
+@NgModule({
+  declarations: [
+    AppComponent,
+    TodoListComponent
+  ],
+  imports: [
+    ...
+  ],
+  providers: [
+    { provide: 'StreamContract', useFactory: ()=> Telescope.of( AppState.empty() ), deps: [] }
+  ],
+  bootstrap: [AppComponent]
+})
+export class AppModule { }
+```
+
+We define our global STATE (however, the more ultimate approach would be to have local state)
+local state = would mean each Service, would have a Telescope instance of their own, resulting in us not 
+having to setup and/or maintain <AppState> 
+Note: If choose to maintain <AppState> as in this example, we would be adding more members to the 
+class, which would mean boilerplate. 
+
+
+```typescript
+interface STATE {
+    todos: TodoState;
+    people: PersonsState;
+}
+
+
+export class AppState implements STATE {
+
+    static empty() {
+        return new AppState( 
+            new TodoState(), 
+            new PersonsState() 
+            );
+    }
+    todos: TodoState;
+    people: PersonsState;
+    
+    constructor(todos: TodoState, people: PersonsState ) {
+        this.todos = todos;
+        this.people = people;
+    }
+}
+
+
+export class TodoState {
+    todos: Array<Todo>;
+
+    constructor( todos: Array<Todo> = [] ) {
+        this.todos = todos;
+    }
+
+    add(title: string): TodoState {
+        let todos = [ {title } ];
+        return new TodoState( [...this.todos, ...todos] );
+    }
+}
+
+export class Todo {
+    constructor( public title: string = ''){}
+}
+
+export class PersonsState {
+    people: Array<Person>;
+}
+
+
+class Person {
+    name: string;
+}
+```
+
+Setup a Service - that will use `StreamContract`
+
+```typescript 
+export class TodoService {
+  state$: Telescope<TodoState>;
+  loaded: boolean = false;
+  constructor(@Inject('StreamContract') private streamContract: Telescope<AppState>) { 
+    /*
+      this will allow us to slice the GLOBAL state 
+      so we can only work with TODOs 
+      since in the AppState  has people
+    */
+    this.state$ = this.streamContract.magnify( todosLens() );  
+  }
+
+  load() {
+    if(!this.loaded) {
+      //httpCall - on success - we get these from server 
+      //[new Todo('go to aroma'),new Todo('fix tires')]
+      let fromServer = [new Todo('go to aroma'),new Todo('fix tires')];
+      loadTodos(this.streamContract, fromServer);
+      this.loaded = true;
+    }
+  }
```

Comment:
Can be `loaded` be part of the state? I'm not sure I'm understanding this example. It seems to me that you are showing how to use Streams more than Telescopes. I think that this example is really good for a sample repo or a post on its own, but for a snippet on this Readme, I think we should look for a minimal sample that shows the basics. We should add link here to a complete post if you decide to write one.

Reply:
I just annotate the HTTP calls - which will feed into the telescope. 

When HTTP succeeds - we take that data and feed it into telescope.

Is there another way?